span-stacktrace/src/main/java/io/opentelemetry/contrib/stacktrace/util/AbstractSimpleChainingSpanProcessor.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.stacktrace.util;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.Arrays;
+
+/**
+ * A @{@link SpanProcessor} which in addition to all standard operations is capable of modifying and
+ * optionally filtering spans in the end-callback.
+ *
+ * <p>This is done by chaining processors and registering only the first processor with the SDK.
+ * Mutations can be performed in {@link #doOnEnd(ReadableSpan)} by wrapping the span in a {@link
+ * MutableSpan}
+ */
+public abstract class AbstractSimpleChainingSpanProcessor implements SpanProcessor {
```

Comment:
are the util package classes part of the public API? if not, we use the convention of putting them somewhere under an `.internal.` package

Reply:
For now, they aren't meant to be public, so we can definitely move them to an `.internal.` package. We would need to make them public only in the case where we need to have another implementation of it in a separate module, but there is no such yet.

In practice it also depends on the progress we might see on having a beforeEnd span processor hook, but that's definitely not granted nor in any known time frame, so safer.

So, to summarize I will rename `...contrib.stacktrace.util` package to `...contrib.stacktrace.internal` to fit this convention.

Reply:
Fixed in 7f09b322ed2f43c66e7c8f3cebbeaa149666d090